provider/aws/aws_test.go

```diff
@@ -1875,13 +1877,37 @@ func TestAWSisAWSAlias(t *testing.T) {
 func TestAWSCanonicalHostedZone(t *testing.T) {
 	for suffix, id := range canonicalHostedZones {
 		zone := canonicalHostedZone(fmt.Sprintf("foo.%s", suffix))
-		assert.Equal(t, id, zone)
+		assert.Equal(t, id, zone, "zone suffix: %s", suffix)
 	}
 
 	zone := canonicalHostedZone("foo.example.org")
 	assert.Equal(t, "", zone, "no canonical zone should be returned for a non-aws hostname")
 }
 
+func TestAWSCanonicalHostedZoneNotExist(t *testing.T) {
+	var buf bytes.Buffer
+	log.SetOutput(&buf)
+	host := "foo.elb.eastwest-1.amazonaws.com"
+	_ = canonicalHostedZone(host)
+	assert.Containsf(t, buf.String(), "Could not find canonical hosted zone for domain", host)
+}
+
+func BenchmarkTestAWSCanonicalHostedZone(b *testing.B) {
+	for i := 0; i < b.N; i++ {
+		for suffix, _ := range canonicalHostedZones {
+			_ = canonicalHostedZone(fmt.Sprintf("foo.%s", suffix))
```

Comment:
I am not sure to understand this test. I'm probably missing something. Should not it check what is returned ?

Reply:
This test does not validate the correctness, but benchmarking actual execution. If we validate here for with assert or smth, means it's an extra operation, I could add it with no issues. But for correctness there is a normal tests, like TestAWSCanonicalHostedZone or TestAWSCanonicalHostedZoneNotExist. 

Benchmarking tests primarily focus on measuring the execution time of the code.

Not sure if there is a good explanation, I could try to google a bit more if needed https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go

Reply:
Do you want me to add 
```
r := canonicalHostedZone(fmt.Sprintf("foo.%s", suffix))
```

In case you think that compiler could eliminate a function call?

Reply:
No, it's fine. I was not aware of that feature in Go. 
Thanks :pray: